tools/llm/run_vlm.py

```diff
@@ -0,0 +1,387 @@
+"""
+.. _run_vlm:
+
+Running VLM inference with Torch-TensorRT
+==========================================================
+
+This script mirrors the style and structure of *run_llm.py*, illustrating a
+Torch-TensorRT (dynamo backend) workflow for Visual-Language Models (VLMs).
+"""
+
+import argparse
+import copy
+import os
+import sys
+from contextlib import nullcontext
+from typing import Tuple
+
+import requests
+import torch
+import torch_tensorrt
+from PIL import Image
+from torchtrt_ext import register_sdpa
+from transformers import AutoModel, AutoProcessor
+from utils import (
+    generate_mm,
+    generate_mm_with_static_cache,
+    record_stats,
+    time_generate_mm,
+)
+
+# -----------------------------------------------------------------------------#
+# Global configuration
+# -----------------------------------------------------------------------------#
+DEVICE = torch.device("cuda:0")
+
+# Register SDPA as a standalone operator.  Converter & lowering pass are defined
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+import transformers.models.qwen2.modeling_qwen2 as mq  # noqa: E402
+
+mq.ALL_ATTENTION_FUNCTIONS["flash_attention_2"] = mq.ALL_ATTENTION_FUNCTIONS["sdpa"]
+
+# -----------------------------------------------------------------------------#
+# Model loading helpers
+# -----------------------------------------------------------------------------#
+
+
+def _load_eagle2(device: torch.device, torch_dtype: torch.dtype):
+    """
+    Load Eagle2 model and processor.
+
+    Returns
+    -------
+    tuple[torch.nn.Module, transformers.AutoProcessor, torch.nn.Embedding]
+        The model, its processor and the language-model input embedding layer.
+    """
+    model_id = "nvidia/Eagle2-2B"
+    with torch.no_grad():
+        model = (
+            AutoModel.from_pretrained(
+                model_id, trust_remote_code=True, torch_dtype=torch_dtype
+            )
+            .eval()
+            .to(device)
+        )
+
+    processor = AutoProcessor.from_pretrained(
+        model_id, trust_remote_code=True, use_fast=True
+    )
+    if hasattr(processor, "tokenizer"):
+        processor.tokenizer.padding_side = "left"
+
+    emb_layer = model.language_model.get_input_embeddings().to(torch_dtype).to(device)
+    return model, processor, emb_layer
+
+
+def _load_model(
+    model_name: str, device: torch.device, torch_dtype: torch.dtype
+) -> Tuple[torch.nn.Module, AutoProcessor, torch.nn.Embedding]:
+    """Dispatch helper for supported VLMs."""
+    if model_name.lower() == "eagle2":
+        return _load_eagle2(device, torch_dtype)
+    msg = f"Unsupported model: {model_name}"
```

Comment:
Consider modifying the message to Encountered model: {model_name} is not supported. Supported models include nvidia/Eagle2-2B, PaliGemma2 (and others if they are supported)